packages/preview-middleware/templates/test/qunit.js

```diff
@@ -0,0 +1,6 @@
+sap.ui.define([
+    'sap/ui/thirdparty/qunit-2'
+], function() {
+        sap.ui.require(<%- JSON.stringify(tests) %>);
```

Comment:
There's a chance to run into a race condition when QUnit is loaded and already looks for tests, but the test modules are not completely loaded, yet. It might cause that none or not all tests are run.

To prevent such scenarios, the code should be adjusted like this (untested):
```js
sap.ui.define([
    'sap/ui/thirdparty/qunit-2'
], function(QUnit) {
    QUnit.config.autostart = false;
    sap.ui.require(<%- JSON.stringify(tests) %>, function() {
        QUnit.start();
    });
});

Reply:
Fill be fixed with https://github.com/SAP/open-ux-tools/pull/1813